pkgs/by-name/ac/actual-server/package.nix

```diff
@@ -0,0 +1,121 @@
+{
+  lib,
+  stdenv,
+  stdenvNoCC,
+  fetchFromGitHub,
+  makeWrapper,
+  cacert,
+  gitMinimal,
+  nodejs_20,
+  yarn,
+  nixosTests,
+  nix-update-script,
+}:
+let
+  version = "24.12.0";
+  src = fetchFromGitHub {
+    owner = "actualbudget";
+    repo = "actual-server";
+    tag = "v${version}";
+    hash = "sha256-qCATfpYjDlR2LaalkF0/b5tD4HDE4aNDrLvTC4g0ctY=";
+  };
+
+  yarn_20 = yarn.override { nodejs = nodejs_20; };
+
+  # We cannot use fetchYarnDeps because that doesn't support yarn2/berry
+  # lockfiles (see https://github.com/NixOS/nixpkgs/issues/254369)
+  offlineCache = stdenvNoCC.mkDerivation {
+    name = "actual-server-${version}-offline-cache";
+    inherit src;
+
+    nativeBuildInputs = [
+      cacert # needed for git
+      gitMinimal # needed to download git dependencies
+      yarn_20
+    ];
+
+    SUPPORTED_ARCHITECTURES = builtins.toJSON {
+      os = [
+        "darwin"
+        "linux"
+      ];
+      cpu = [
+        "arm"
+        "arm64"
+        "ia32"
+        "x64"
+      ];
+      libc = [
+        "glibc"
+        "musl"
+      ];
+    };
+
+    buildPhase = ''
+      runHook preBuild
+
+      export HOME=$(mktemp -d)
+      yarn config set enableTelemetry 0
+      yarn config set cacheFolder $out
+      yarn config set --json supportedArchitectures "$SUPPORTED_ARCHITECTURES"
+      yarn
```

Comment:
Unfortunately this will only build the native libraries for the current platform, meaning the content hash will vary (as seen by ofborg).

I already tried to fix it (by building in the package derivation), but couldn't make it work.

Reply:
The solution would be to use `fetchYarnDeps` but we can't since it doesn't support v2/berry lockfiles yet. So i guess the best workaround we have is to set the output hash conditionally, at least this is what is done in the grafana build which also does that.

I guess this will make updates a bit more tedious, but we probably have no other choice.